server/resolvers/TestResult/unexpectedBehaviorCountResolver.js

```diff
@@ -0,0 +1,11 @@
+const unexpectedBehaviorCount = testResult => {
+    const { result } = testResult;
+    const { details } = result;
+    const { summary } = details;
+
+    if (!details || !summary) return 0;
+
+    return summary.unexpectedCount;
```

Comment:
The database values are quite different from the GraphQL values and I'm worried this could get confusing! Is this something you think we can clean up as we go or do think we can clean it up before we merge?

Reply:
Actually, these are the same variables currently stored for our examples in the `TestPlanRun.testResults` table columns. They were originally ported over from the old implementation. This object structure and variables come directly from the result of the iframe's result when submitted.
What are your suggestions here?

Reply:
I think we should design an elegant data structure that fits nicely into GraphQL, which I think is what we're going to do with the TestResultInput type, and then I think we have two options about how to handle the old model:

First, we can implement [an anticorruption layer](https://dev.to/asarnaout/the-anti-corruption-layer-pattern-pcd), basically a remapping function takes an instance of the old model and returns the same data but organized in the same way as the GraphQL model. We do that immediately, probably on the frontend, or as soon as possible, to avoid using the old model in any code. The idea there is that having inaccurate or out-of-date models will make the terminology less meaningful, or "corrupt" it.

Second, and this is probably what we have to do eventually, but is probably too difficult to do immediately, is actually rewrite the system so that it produces data in the correct format without any remapping needed.

Reply:
> First, we can implement an anticorruption layer, basically a remapping function takes an instance of the old model and returns the same data but organized in the same way as the GraphQL model. We do that immediately, probably on the frontend, or as soon as possible, to avoid using the old model in any code. The idea there is that having inaccurate or out-of-date models will make the terminology less meaningful, or "corrupt" it.

Do you mean doing something like this? https://github.com/w3c/aria-at-app/blob/614f9234eb218c8aec26688b7c79bfb02768fbde/server/util/validateTestResult.js#L58

> Second, and this is probably what we have to do eventually, but is probably too difficult to do immediately, is actually rewrite the system so that it produces data in the correct format without any remapping needed.

Agreed that it could come to that but too big of an ask at the current moment

Reply:
> I think we should design an elegant data structure that fits nicely into GraphQL, which I think is what we're going to do with the TestResultInput type

@alflennik and I experimented with this and it proved to be too much effort currently